transactionoutbox-cdi/src/main/java/com/gruelbox/transactionoutbox/CdiInstanciator.java

```diff
@@ -0,0 +1,15 @@
+package com.gruelbox.transactionoutbox;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.inject.spi.CDI;
+
+@ApplicationScoped
+public class CdiInstanciator extends AbstractFullyQualifiedNameInstantiator
+{
+
+   @Override
+   protected Object createInstance(Class<?> clazz)
+   {
+      return CDI.current().select(clazz).get();
```

Comment:
Is there any situation where you might not want to use `CDI.current()`?

Reply:
OK,so it sounds like you could provide the option to do:

```
@Produces
public CdiInstantiator cdiInstantiator(BeanManager bm) {
  return CdiInstantiator.create(bm);
}
```
And thus avoid the need to use the static, but that could easily be added later so no need to worry about it for now.